### PR TITLE
Parameterize MDSD account and namespace names

### PR DESCRIPTION
**Pull request by zacharyljones**
Which issue this PR addresses:
[USER STORY 10098170 - Parameterize Geneva Namespace Targets in ARO RP](https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/10098170/)

What this PR does / why we need it:
This PR allows the RP to read in additional mdsd logging parameters from RP Config files. These new parameters include mdsd log account name, namespace name, and metrics account name for both customer clusters and the rp itself. This was necessary because there are not separate mdsd endpoints for FF INT and PROD. These changes allow different accounts/namespaces to be targeted for different environments at the same mdsd endpoint.

RP Config has already been updated with the name parameters for each environment.

Test plan for issue:
I tested this by deploying these RP changes to FF INT and verifying that the RP log and customer cluster data made it into the Geneva log accounts and namespaces specified in the FF INT config file.

[Build](https://dev.azure.com/msazure/AzureRedHatOpenShift/_build/results?buildId=44464312&view=results) | [Release](https://dev.azure.com/msazure/AzureRedHatOpenShift/_apps/hub/ms.vss-releaseManagement-web.cd-release-progress?_a=release-pipeline-progress&releaseId=806) | [Ev2](https://ev2portal.azure.net/#/Rollout/AroRpDeployTest/e0700e45-afed-43a3-b319-9f400b23e583?RolloutInfra=Fairfax)

Is there any documentation that needs to be updated for this PR?
No. RP Config has already been updated with an explanation of all new parameters.

### CODE REVIEW

**Comment by zacharyljones** [comment]
@jewzaam @troy0820 The commit history for [PR-1601](https://github.com/Azure/ARO-RP/pull/1601) was a bit of a mess, so I abandoned that PR in favor of this one. This PR does not have any changes other than the ones directly relevant to this change.

I will copy over the outstanding comments from the previous PR so that they can be addressed here.

File: pkg/deploy/generator/resources_rp.go [hunk 1]
```diff
@@ -622,14 +630,23 @@ ExecStart=/usr/bin/docker run \
...
   -e MDM_NAMESPACE \
   -e MDSD_ENVIRONMENT \
   -e RP_FEATURES \
+  -e AZURE_EV2 \
+  -e AZURE_ENVIRONMENT \
+  -e AZURE_SUBSCRIPTION_ID \
+  -e AZURE_TENANT_ID \
+  -e LOCATION \
+  -e RESOURCEGROUP \
+  -e HOSTNAME_OVERRIDE \
```
Comment (jewzaam) [inline pkg/deploy/generator/resources_rp.go:649]: This is an unrelated change, is it an outstanding patch CF is carrying?

**Review by troy0820 — CHANGES_REQUESTED** [review]
Hold until changes from https://github.com/Azure/ARO-RP/pull/1601 are addressed

**Comment by zacharyljones** [comment]
@troy0820 I may be misinterpreting your vendor comment, but I am responsible for all changes in this PR. There is no need to break out any particular piece into a separate commit.

**Comment by troy0820** [comment]
@zacharyljones  We usually break vendor commits (changes that change /vendor folder) into its own commit so they will stand alone from the changes introduced into the repo.  This helps with git history and allows us to target changes introduced by commits/PRs by dependencies, etc.

**Comment by zacharyljones** [comment]
@troy0820 Ah understood - changes to the vendor folder, not actual vendors.

File: pkg/deploy/generator/resources_rp.go (hunk 1 above)
Comment (mikeandescavage) [reply pkg/deploy/generator/resources_rp.go:649]: This ultimately came from Rogerio's branch when testing e2e variables (https://github.com/rogbas/ARO-RP/blob/d292f20ce01673b6a9743aa2ddfbb2981e7ef862/pkg/deploy/generator/resources_rp.go#L633)

This is not related to this PR, however, we need to reconvene on all the E2E changes

Comment (zacharyljones) [reply pkg/deploy/generator/resources_rp.go]: This block has been removed and newly generated files have been added

File: pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go [hunk 2]
```diff
@@ -30,8 +30,12 @@ func ClusterChecksTypes() []status.ConditionType {
 type GenevaLoggingSpec struct {
 	// +kubebuilder:validation:Pattern:=`[0-9]+.[0-9]+`
 	ConfigVersion string `json:"configVersion,omitempty"`
-	// +kubebuilder:validation:Enum=DiagnosticsProd;Test
```
Comment (zacharyljones) [inline pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go:33]: @troy0820 - "I'm not sure why this is removed is the kubebuilder:validation:Enum=DiagnosticsProd,Test;CaFairFax is also on line 35"

Comment (zacharyljones) [reply pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go:33]: I'm not sure either - may just be a caveat of how I copied and pasted to create the new parameters. The appropriate kubebuilder tags are there though.

File: pkg/util/mocks/env/env.go [hunk 3]
```diff
@@ -407,6 +421,20 @@ func (mr *MockInterfaceMockRecorder) FeatureIsSet(arg0 interface{}) *gomock.Call
...
+	ret0, _ := ret[0].(string)
+	return ret0
+}
+
+// GenevaLoggingEnvironment indicates an expected call of GenevaLoggingEnvironment.
+func (mr *MockInterfaceMockRecorder) GenevaLoggingEnvironment() *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "GenevaLoggingEnvironment", reflect.TypeOf((*MockInterface)(nil).GenevaLoggingEnvironment))
+}
+
```
Comment (zacharyljones) [inline pkg/util/mocks/env/env.go:437]: @troy0820 - "[ ] Update unit test to include new (generation of new mocks have been made and existing unit tests do not cover newly included config params, etc)"

Comment (zacharyljones) [reply pkg/util/mocks/env/env.go:437]: From what I can tell, existing unit tests do not cover anything related to mdsd logging. I agree that unit tests should be added, but feel that creating a brand new suite of tests for mdsd logging is beyond the scope of this PR/work item.

**Comment by zacharyljones** [comment]
@troy0820 All changes to the vendor folder have been moved to a separate commit.

File: pkg/env/prod.go [hunk 4]
```diff
@@ -91,8 +95,10 @@ func newProd(ctx context.Context, log *logrus.Entry) (*prod, error) {
 
 		fpClientID: os.Getenv("AZURE_FP_CLIENT_ID"),
 
-		clusterGenevaLoggingEnvironment:   os.Getenv("MDSD_ENVIRONMENT"),
+		clusterGenevaLoggingAccount:       os.Getenv("CLUSTER_MDSD_ACCOUNT"),
+		clusterGenevaLoggingNamespace:     os.Getenv("CLUSTER_MDSD_NAMESPACE"),
 		clusterGenevaLoggingConfigVersion: os.Getenv("CLUSTER_MDSD_CONFIG_VERSION"),
+		genevaLoggingEnvironment:          os.Getenv("MDSD_ENVIRONMENT"),
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply pkg/env/prod.go:101]: I think this make sense.

File: pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go (hunk 2 above)
Comment (mjudeikis) [reply pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go:33]: It is bellow : `// +kubebuilder:validation:Enum=DiagnosticsProd;Test;CaFairfax`. So change expected

File: vendor/github.com/openshift/installer/pkg/asset/bootstraplogging/bootstraplogging.go [hunk 5]
```diff
@@ -12,6 +12,7 @@ const (
 type Config struct {
 	Certificate       string
 	Key               string
+	Account           string
```
Comment (mjudeikis) [inline vendor/github.com/openshift/installer/pkg/asset/bootstraplogging/bootstraplogging.go:15]: No no :) Lets not do this :)

Please fix a change in `https://github.com/mjudeikis/installer/blob/release-4.7-azure`.
You will need to get this branch check out, find commit which adds bootstrap logging and "edit" individual commit to add this change in there and make a PR into our Installer fork. 

This not gonna fly :)

https://github.com/mjudeikis/installer/blob/release-4.7-azure/pkg/asset/bootstraplogging/bootstraplogging.go#L14

**Review by mjudeikis — CHANGES_REQUESTED** [review]

File: pkg/env/prod.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (troy0820) [reply pkg/env/prod.go:101]: >The reason I changed the name of this variable is because even though there are two sets of parameters for this change (cluster mdsd and rp mdsd), the genevaLoggingEnvironment parameter is shared between the two different configurations.

Does this make sense to move this to `pkg/env/core.go`?  Only because looking at the prod struct I expect to see things that are consistent with the cluster and not shared between the rp and the cluster as far as configuration.  I understand that the only instance of this is in this prod struct but explicitly sharing this configuration from the cluster and the RP isn't something I'd expect to see when this struct is used for prod cluster configuration.

[the remark this answers is not in the document]
Comment (zacharyljones) [reply pkg/env/prod.go:101]: I think that's a fair point.

[the remark this answers is not in the document]
Comment (zacharyljones) [reply pkg/env/prod.go:101]: @mjudeikis @troy0820 I'm going to change the name back.

[the remark this answers is not in the document]
Comment (zacharyljones) [reply pkg/env/prod.go:101]: I would rather do that than move anything

File: vendor/github.com/openshift/installer/pkg/asset/bootstraplogging/bootstraplogging.go (hunk 5 above)
Comment (zacharyljones) [reply vendor/github.com/openshift/installer/pkg/asset/bootstraplogging/bootstraplogging.go]: @mjudeikis I reverted this particular change - parameterizing the log account name in the bootstrapper logging is beyond the scope of this change.

File: pkg/env/prod.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (zacharyljones) [reply pkg/env/prod.go]: @troy0820 - I went ahead and changed the names back for everything except the reference to variables in const.go

**Comment by zacharyljones** [comment]
Still getting Cluster and RP logs in Geneva after deployment of latest round changes:

[Build](https://dev.azure.com/msazure/AzureRedHatOpenShift/_build/results?buildId=44499725&view=results)

[Release 294](https://dev.azure.com/msazure/AzureRedHatOpenShift/_releaseProgress?_a=release-pipeline-progress&releaseId=813)

[Ev2 Rollout](https://ev2portal.azure.net/#/Rollout/AroRpDeployTest/271f5755-bb40-46a6-98af-e2bc793f350a?RolloutInfra=Fairfax)

File: vendor/github.com/openshift/installer/pkg/asset/bootstraplogging/bootstraplogging.go (hunk 5 above)
Comment (mjudeikis) [reply vendor/github.com/openshift/installer/pkg/asset/bootstraplogging/bootstraplogging.go]: But how this will work in Bootstrap logs in FF? This will break it?

**Comment by jewzaam** [comment]
Discussed and we'll get this PR merged and address the outstanding concern / issue with bootstrap node logs in a separate  PR.

**Review by jewzaam — APPROVED** [review]
